### PR TITLE
guide: update exp sharing guide

### PR DESCRIPTION
**Pull request by dberenbaum**
Updates to experiments sharing guide now that it is close to feature complete. Tried to cover it as one holistic end-to-end workflow, including:

* Trying to explain use cases for live metrics vs exp refs vs persistent commits and when and why you might want each
* Emphasize where Studio fits and how it helps
* Include features that aren't strictly sharing but are part of the overall workflow (finding shared experiments, persisting, cleaning)
* Deemphasize explaining all the different options in the CLI

### CODE REVIEW

**Comment by shcheklein** [comment]
Great update, thanks @dberenbaum !

There is the whole section - https://dvc-org-exp-sharing-lz1qsypgwi.herokuapp.com/doc/user-guide/experiment-management/persisting-experiments - do we still need it? The same for cleaning experiments section.

**Comment by skshetry** [comment]
Sorry, accidentally closed. :)

**Comment by dberenbaum** [comment]
> There is the whole section - https://dvc-org-exp-sharing-lz1qsypgwi.herokuapp.com/doc/user-guide/experiment-management/persisting-experiments - do we still need it? The same for cleaning experiments section.

Yup, sorry for not communicating any sort of holistic plan here. My hope is to merge:

- #4606 
- #4650 (this PR)
- #4651 

Then to drop those pages you mention and consolidate the experiments guide.

**Review by shcheklein — APPROVED** [review]

File: content/docs/user-guide/experiment-management/sharing-experiments.md [hunk 1]
```diff
@@ -166,12 +205,37 @@ share these, `dvc push` them to [remote storage] (e.g. Google Drive or NAS).
...
+## Remove pushed experiments
+
+As you share more experiments, [Studio] and Git remotes may be become cluttered
+with experiment references.
+
+You can remove experiments in Studio:
+
+![Studio Remove selected rows](/img/studio-remove.gif)
+
+To remove experiments using the command line, use `dvc exp remove -g`:
```
Comment (dberenbaum) [inline content/docs/user-guide/experiment-management/sharing-experiments.md:221]:
```suggestion
To remove pushed experiments using the command line, use `dvc exp remove -g`:
```

**Pull request merged**